include/SH3/graphics/msbmp.hpp

```diff
@@ -0,0 +1,130 @@
+/** @file
+ *
+ *  Class and data structures to load Microsoft BMP/DIB files.
+ *
+ *  @copyright 2016-2017  Palm Studios
+ *
+ *  @date 6-2-2017
+ *
+ *  @note All paths specified by the constructor are PHYSICAL paths on disk.
+ *  @note The palette data is stored as BGR with the last byte being 0xFF (unused)
+ *
+ *  @author Jesse Buhagiar
+ */
+#ifndef MSBMP_HPP_INCLUDED
+#define MSBMP_HPP_INCLUDED
+
+#include "SH3/error.hpp"
+
+#include <cstdint>
+#include <vector>
+#include <string>
+
+#include <GL/gl.h>
+
+namespace sh3_graphics
+{
+    /** @defgroup graphics-headers Graphics headers
+     *  @{
+     */
+
+     #pragma pack(push, 1) // Stop the compiler from giving us stupid naturally aligned boundaries (__attribute__((packed)) didn't work...)
```

Comment:
Interesting. What do you get running this program: https://gist.github.com/z33ky/5ba9cf15d02615e62ae7927afaf1728d